glossary/Function.md

```diff
@@ -0,0 +1,5 @@
+### Function
+
+Functions are self-contained blocks of code with their own scope, that can be called by other code and are usually associated with a unique identifier.
+Functions accept input in the form of arguments and can optionally return an output. 
+JavaScript functions are also objects.
```

Comment:
could amend it here as well

Reply:
For this term, I think stating that they're object is sufficient..

Reply:
> Functions are special objects in JavaScript that have their own local scope and can be called by other pieces of code. They are usually assigned a unique identifier, but sometimes an anonymous function is used. Functions can optionally accept inputs and return outputs.

Combines both but may be confusing. It should also be noted that a Function call will always return. If one is not explicit, the implicit return of `undefined` is returned.

Reply:
I prefer the phrasing asis. The return functionality, however, should be explained. I'll update.